File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/server/messages/ServerMessageAddDocument.java
package net.stzups.board.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.Document;
import net.stzups.board.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageAddDocument extends ServerMessage {
    private Document document;

    public ServerMessageAddDocument(Document document) {
        super(ServerMessageType.ADD_DOCUMENT);
        this.document = document;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeLong(document.getId());
        writeString(document.getName(), byteBuf);
    }
}

File: frontend/html/scripts/protocol/server/messages/ServerMessageAddDocument.js
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";

export default class ServerMessageAddDocument extends ServerMessage {
    constructor(reader) {
        super(ServerMessageType.ADD_DOCUMENT);
        this.id = reader.readBigInt64();
        this.name = reader.readString();
    }
}